src/main/java/com/r307/arbitrader/service/ExchangeService.java

```diff
@@ -283,19 +304,40 @@ public BigDecimal getExchangeFee(Exchange exchange, CurrencyPair currencyPair, b
                 }
 
                 // give up and return a default fee - this is intentionally higher than most exchanges
-                return new BigDecimal("0.0030");
+                return new ExchangeFee(shortFee, new BigDecimal("0.0030"));
             }
 
             if (!isQuiet) {
                 LOGGER.warn("{} fees unavailable via API. Will use configured value.",
                     exchange.getExchangeSpecification().getExchangeName());
             }
 
-            return configuredFee;
+            return new ExchangeFee(shortFee, configuredFee);
         }
 
         // Last fall back - use CurrencyPairMetaData trading fee
-        feeCache.setCachedFee(exchange, currencyPair, currencyPairMetaData.getTradingFee());
-        return currencyPairMetaData.getTradingFee();
+        final ExchangeFee exchangeFee = new ExchangeFee(shortFee, currencyPairMetaData.getTradingFee());
+        feeCache.setCachedFee(exchange, currencyPair, exchangeFee);
+        return exchangeFee;
+    }
+
+    @Nullable
+    private BigDecimal getShortFee(ExchangeConfiguration exchangeMetadata) {
+        final BigDecimal shortFee;
+
+        // If exchange is set for margin trade then get the configured margin (short) fee
+        if (exchangeMetadata.getMargin()) {
+            if (exchangeMetadata.getMarginFeeOverrride() != null) {
+                shortFee = exchangeMetadata.getMarginFeeOverrride();
```

Comment:
Still a couple extra 'r's :)

Reply:
Rrrrrrr! :D

Reply:
Hmm something is wrong here. I don't have that extra 'r' in my code and I have pushed everything :thinking: 
Maybe you are seeing an old commit?

This is how the code is on my side:

`@Nullable
private BigDecimal getShortFee(ExchangeConfiguration exchangeMetadata) {
    final BigDecimal shortFee;

    // If exchange is set for margin trade then get the configured margin (short) fee
    if (exchangeMetadata.getMargin()) {
        if (exchangeMetadata.getMarginFeeOverride() != null) {
            shortFee = exchangeMetadata.getMarginFeeOverride();
        }
        else {
            shortFee = exchangeMetadata.getMarginFee();
        }
    }
    else {
        shortFee = null;
    }

    return shortFee;
}`

Searching for `getMarginFeeOverrride` returns no results.

Reply:
It's on your branch on GitHub. It's possible you have changes on your computer that haven't been pushed?

https://github.com/dazito/arbitrader/blob/bootstap-margin-fee-exch-configuration/src/main/java/com/r307/arbitrader/service/ExchangeService.java#L331